src/d3d9/d3d9_device.cpp

```diff
@@ -3992,7 +3992,21 @@ namespace dxvk {
 
       if (alloced)
         std::memset(physSlice.mapPtr, 0, physSlice.length);
-      else if (!skipWait) {
+      else if (managed && !skipWait) {
+        if (!WaitForResource(mappedBuffer, D3DLOCK_DONOTWAIT)) {
```

Comment:
maybe we should take lock flags out of WaitForResource and just make it a BOOL

Reply:
Would probably make sense to rename the function as well if we do that, given that it doesn't really wait every time.

Reply:
ye but we can do that after